sqlmesh/core/dialect.py

```diff
@@ -755,12 +755,23 @@ def select_from_values_for_batch_range(
         exp.alias_(exp.cast(column, to=kind), column, copy=False)
         for column, kind in columns_to_types.items()
     ]
-    values_exp = exp.values(
-        [tuple(transform_values(v, columns_to_types)) for v in values[batch_start:batch_end]],
-        alias=alias,
-        columns=columns_to_types,
-    )
-    return exp.select(*casted_columns).from_(values_exp, copy=False)
+    select = exp.select(*casted_columns)
+
+    # Make sure we don't generate an empty VALUES clause
+    if values:
+        values_exp = exp.values(
+            [tuple(transform_values(v, columns_to_types)) for v in values[batch_start:batch_end]],
+            alias=alias,
+            columns=columns_to_types,
+        )
+        return select.from_(values_exp, copy=False)
+
+    # If there are no values, return a dummy query that always produces no rows
+    for projection in select.selects:
```

Comment:
select.set("expressions", [exp.Literal.number(1)])

Reply:
ah you need the same number? in the case, it should be nulls

Reply:
select.set("expressions", [exp.null().as_(alias) for alias in select.named_selects])

Reply:
Got it makes sense, yeah I need the same number to match the aliases in things like create view etc